File: src/programmers/lv2_printer.java
package programmers;

import java.util.Collections;
import java.util.PriorityQueue;

public class lv2_printer {
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] arr = {2,1,3,4};
        System.out.print(solution.solution(arr,2));
    }

    static class Solution {
        public int solution(int[] priorities, int location) {
            PriorityQueue<Integer> queue = new PriorityQueue<>(Collections.reverseOrder());
            int answer = 0;

            for (int i = 0; i < priorities.length; i++) queue.add(priorities[i]);

            while (!queue.isEmpty()) {
                for (int i = 0; i < priorities.length; i++) {
                    if (priorities[i] == queue.peek()) {
                        answer++;
                        if (i == location) {
                            return answer;
                        }
                        queue.poll();
                    }
                }
            }
            return -1;
        }
    }
}

File: JavaKotlin/src/lv2_printer.kt
import java.util.*


fun main() {
    val solution3 = Solution3()
    print(solution3.solution(intArrayOf(2, 1, 3, 2), 2))
}

class Solution3 {
    fun solution(priorities: IntArray, location: Int): Int {
        val priorityQueue = PriorityQueue(Collections.reverseOrder<Int>())
        var answer = 0

        for (element in priorities) priorityQueue.add(element)

        while (!priorityQueue.isEmpty()) {
            for (i in priorities.indices) {
                if (priorities[i] === priorityQueue.peek()) {
                    if (i == location) {
                        answer++
                        return answer
                    }
                    priorityQueue.poll()
                    answer++
                }
            }
        }
        return -1
    }
}